features/bsm/service/impl/src/test/java/org/opennms/netmgt/bsm/service/internal/BusinessServiceManagerImplIT.java

```diff
@@ -153,6 +159,65 @@ public void testGetOperationalStatusForBusinessService() {
+        BusinessService service3 = createService("Business Service #3");
+
+        Long serviceId1 = businessServiceDao.save(service1);
+        Long serviceId2 = businessServiceDao.save(service2);
+        Long serviceId3 = businessServiceDao.save(service3);
+
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId2),
+                                            businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId1)));
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId1),
+                                            businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId2)));
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId1),
+                                            businessServiceManager.getById(serviceId2)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId3)));
+
+        businessServiceManager.assignChildService(serviceId1, serviceId2);
+
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId2),
+                                            businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId1)));
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId2)));
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId1),
+                                            businessServiceManager.getById(serviceId2)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId3)));
+
+        businessServiceManager.assignChildService(serviceId2, serviceId3);
+
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId2),
+                                            businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId1)));
+        Assert.assertEquals(ImmutableSet.of(businessServiceManager.getById(serviceId3)),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId2)));
+        Assert.assertEquals(ImmutableSet.of(),
+                            businessServiceManager.getFeasibleChildServices(businessServiceManager.getById(serviceId3)));
+    }
+
+    @Test
+    public void testLoopCreation() {
+        thrown.expect(IllegalArgumentException.class);
```

Comment:
I don't think new exception types are a good idea - but we should discuss this somewhere else. To ensure the right exception is triggered, I will check themessage, too.